### ISSUE TITLE
Privilege Assistant installation failed. I have tried many times and all the results are the same.

### ISSUE THREAD

**Issue by DevLiuSir** [ninxsoft/Mist#69]
Privilege Assistant installation failed. I have tried many times and all the results are the same.
<img width="540" alt="截屏2023-07-02 10 26 02" src="https://github.com/ninxsoft/Mist/assets/11488337/e69f71f6-9455-449b-9e4a-178c846979c7">

**Comment by ninxsoft** [comment]
The Privileged Helper tool installs the following files:

- `/Library/LaunchDaemons/com.ninxsoft.mist.helper.plist`
- `/Library/PrivilegedHelperTools/com.ninxsoft.mist.helper`

The screenshot you provided appears to be failing when updating these two files. You have a couple of options:

1. Run the uninstall script [found here](https://github.com/ninxsoft/Mist/blob/main/Scripts/uninstall.sh), then download the latest Mist **disk image (dmg)** from the [release page](https://github.com/ninxsoft/Mist/releases). Then try to install the Privileged Helper Tool again from the Mist app settings.
2. Download and install the latest Mist **installer package (pkg)** from  the [releases page](https://github.com/ninxsoft/Mist/releases). This will place the files mentioned above into the correct locations.

**Comment by DevLiuSir** [comment]
The latest version 0.9 downloaded, but the installation of the Privilege Assistant failed

**Comment by DevLiuSir** [comment]
Removed using uninstallation script. After reinstalling the app. Failed to install Privilege Assistant again

**Comment by DevLiuSir** [comment]
> Same issue here. Ran the uninstall script, and it cleaned up any reminisce. Tried installing the pkg file, and it errors out. DMG installs, but the app cannot install privileged helper with the same error above.

Did you also fail to install it?

**Comment by bhagatparwinder** [comment]
@DevLiuSir Correct. The .pkg installation fails. I can copy the app from the DMG, but it cannot install the privileged tool. It gives me the same error that you posted a screenshot of.

**Comment by ninxsoft** [comment]
@DevLiuSir + @bhagatparwinder thank you so much for the help with testing 🙏

The package install has three steps, performed in the following order:

1. Run the [preinstall](https://github.com/ninxsoft/Mist/blob/main/Scripts/preinstall) script
2. Copy the contents of the payload to the target disk:
   - `/Applications/Mist.app`
   - `/Library/LaunchDaemons/com.ninxsoft.mist.helper.plist`
   - `/Library/PrivilegedHelperTools/com.ninxsoft.mist.helper`
3. Run the [postinstall](https://github.com/ninxsoft/Mist/blob/main/Scripts/postinstall) script

If steps **1** or **3** fail, the entire package install fails - are you able to run the **preinstall** and **postinstall** scripts manually and report back the error / output message you are receiving?

If we can isolate the specific command in the scripts that is failing, we should be able to move forward with a fix.

Thank you 🙌

**Comment by ninxsoft** [comment]
Having said that, you have both mentioned that attempting to install the Privileged Helper Tool within the Mist.app settings also fails, so I am learning towards seeing if I can make the `BlessError` message more descriptive.

Will keep you posted 👍

**Comment by ninxsoft** [comment]
Please ignore the previous request to test the **preinstall** and **postinstall** scripts - I have improved the error message output for **BlessError** exceptions:

![BlessError](https://github.com/ninxsoft/Mist/assets/4812595/d13bb9f3-8e9c-4460-bf5e-18fe54f0c4ff)

⚠️ **Note:** This is only an example screenshot, so I will request you to test again once I release the next version of **Mist.app** 👍

**Comment by ninxsoft** [comment]
[Mist version 0.9.1](https://github.com/ninxsoft/Mist/releases/tag/v0.9.1) is now available - if you can please test with this version and report back 🙌

**Comment by bhagatparwinder** [comment]
- I downloaded the latest release (DMG file).
- Dragged mist to Applications.
- Started mist app
- It asked to install the helper tool > enter admin password > see a notification that Mist has added something in the background
- Tried a download and it again asks to install the helper tool. Quit the application and started again. Still the same popup.

![SCR-20230825-jfix](https://github.com/ninxsoft/Mist/assets/6809087/e99f10da-61ed-4047-ab6c-b0ddf360d485)

**Comment by ninxsoft** [comment]
@bhagatparwinder how about when you try to install the Privileged Helper Tool via the app settings?

![Before](https://github.com/ninxsoft/Mist/assets/4812595/56df9a7d-ca8a-420f-bf04-08dc7006771b)

![After](https://github.com/ninxsoft/Mist/assets/4812595/589392aa-389c-4cb2-9306-bae5ab071d33)

**Comment by DevLiuSir** [comment]
> 应用程序设置安装特权帮助工具

Application settings installation privileges help tool also error。

<img width="719" alt="截屏2023-08-27 22 52 10" src="https://github.com/ninxsoft/Mist/assets/11488337/4b006f64-93e2-4c91-af36-359d7107e45b">

**Comment by ninxsoft** [comment]
@DevLiuSir thanks for providing the screenshot, we have more error information now!

The error itself does seem confusing though - what happens if you manually delete `/Library/PrivilegedHelperTools/com.ninxsoft.mist.helper` and then try to install the Privileged Helper Tool again from the Settings?

**Comment by DevLiuSir** [comment]
<img width="1155" alt="截屏2023-09-05 15 26 13" src="https://github.com/ninxsoft/Mist/assets/11488337/80ce2f40-d8ef-42e0-b530-e5f21a4cbf83">

Manually delete the daemon. Reinstall, the error message is not present. But it was still unsuccessful. Installation was unsuccessful. Creating a USB drive cannot proceed

**Comment by bhagatparwinder** [comment]
I can attest that the issue exists on 0.9.1 and macOS Sonoma 14.0. I am testing it on ARM.

**Comment by ninxsoft** [comment]
I have improved the error output when attempting to install the Privileged Helper Tool from the main app window.

This is just an example:

<img width="632" alt="Error" src="https://github.com/ninxsoft/Mist/assets/4812595/e1ec67b8-2f3f-4661-a25a-9312614bc8fe">

**See:** #103 - will push a new release soon where we can continue testing 🙏

**Comment by JuanCab** [comment]
Can confirm failure to install privileged helper tools both from Settings and by just having the dialog box trigger when trying to download an installer running macOS 14.2 and Mist 0.9.1.  Manually removed the previous files from disk via command line, still fails.  No error messages, but it doesn't install.
